core/image.cpp

```diff
@@ -1586,11 +1586,12 @@ void Image::create(int p_width, int p_height, bool p_use_mipmaps, Format p_forma
 	ERR_FAIL_COND_MSG(p_height <= 0, "Image height must be greater than 0.");
 	ERR_FAIL_COND_MSG(p_width > MAX_WIDTH, "Image width cannot be greater than " + itos(MAX_WIDTH) + ".");
 	ERR_FAIL_COND_MSG(p_height > MAX_HEIGHT, "Image height cannot be greater than " + itos(MAX_HEIGHT) + ".");
+	ERR_FAIL_COND_MSG(write_lock.ptr(), "Cannot create image when it is locked.");
 
 	int mm = 0;
 	int size = _get_dst_image_size(p_width, p_height, p_format, mm, p_use_mipmaps ? -1 : 0);
-	data.resize(size);
-	{
+
+	if (data.resize(size) == OK) {
```

Comment:
I think this change is out of scope for this PR, and because `master` branch also needs this change, else it will become exclusive to 3.2.

Might as well `ERR_FAIL_COND()` too as seen in some other occurrences of `data.resize()` within the class, because calling `get_width()` and `get_height()` etc would report wrong values, see above. In any case all those occurrences of `resize` should have corresponding error check if required, which might be worth another PR in my eyes.